keyring/backends/chainer.py

```diff
@@ -21,9 +21,11 @@ class ChainerBackend(backend.KeyringBackend):
     @classmethod
     def priority(cls):
         """
-        High-priority if there are backends to chain, otherwise 0.
+        If there are backends to chain, high priority
+        Otherwise very low priority since our operation when empty
+        is the same as null.
         """
-        return 10 * (len(cls.backends) > 1)
```

Comment:
Apologies for the delay in review, and thanks for the contrib.

I'm not sure -10 is meaningful here. Probably -1 should be used. The reason 10 is used on the positive side is because 1 means minimally recommended, 5 is used for more recommended ones, so 10 means preferably to those.

I may adjust this, but it looks good at first blush. Probably there should be a test to protect against the issue.

Reply:
I was also in doubt about −10, but then realized that the idea was to have lower priority than `null` backend, which has −1.

Now I see that you resolved this by using `fail.Keyring.priority - 1`, which is indeed better. Thanks!